network/p2p/error.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import "github.com/ava-labs/avalanchego/snow/engine/common"
+
+var (
+	// ErrUnexpected should be used to indicate that a request failed due to a
+	// generic error
+	ErrUnexpected = &common.AppError{
+		Code:    -1,
```

Comment:
Not sure how to handle this - we decided to reserve negative error codes for our use + positive error codes for developer use. This is a weird intersection since the p2p package is technically a vm implementation detail.